File: Striver_step_1/4. Recursion/problem2.cpp
//print upto n using recursion
#include<bits/stdc++.h>
using namespace std;

void printNN(int cnt,int n){
    if (cnt > n) return;
    cout<<cnt<<endl;
    printNN(++cnt,n);
}
int main(){
    int n;
    cout<<"Enter the no.: ";
    cin>>n;
    printNN(1,n);

    return 0;
}

File: Striver_step_1/4. Recursion/problem2.py
cnt = 1
def printNN(n):
    global cnt;
    if (cnt > n):
        return
    print(cnt)
    cnt += 1
    printNN(n)
    


if __name__ == "__main__":
    n = int(input("Enter the no. : "))
    printNN(n)